### PR TITLE
RavenDB-14909 - avoid modifying the exception hierarchy

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-14909

### CODE REVIEW

File: src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexWriter.cs [hunk 1]
```diff
@@ -73,6 +72,9 @@ public void Commit(IState state)
                 if (e is Win32Exception win32Exception && win32Exception.IsOutOfMemory())
                     ThrowOutOfMemory(e);
 
+                if (e.InnerException is VoronUnrecoverableErrorException)
+                    throw e.InnerException;
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexWriter.cs]: We are now losing the inner stack here.
Maybe throw that in a *good* exception?

File: src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneSuggestionIndexWriter.cs [hunk 2]
```diff
@@ -204,6 +203,9 @@ public void Commit(IState state)
                 if (e is Win32Exception win32Exception && win32Exception.IsOutOfMemory())
                     LuceneIndexWriter.ThrowOutOfMemory(e);
 
+                if (e.InnerException is VoronUnrecoverableErrorException)
+                    throw e.InnerException;
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneSuggestionIndexWriter.cs]: Same

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexWriter.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexWriter.cs]: Do we need the inner stack? (this exception is wrapped)
https://github.com/ravendb/lucenenet/blob/3.0.3-netcore/src/Lucene.Net/Index/IndexWriter.cs#L4644

Comment (ayende) [reply src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexWriter.cs]: We never want to lose the stack

Comment (grisha-kotler) [reply src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexWriter.cs]: done

**Review by ppekrol — APPROVED** [review]

**Pull request merged**